addons/Yush/Yush.lua

```diff
@@ -75,12 +77,24 @@ output = function()
     if settings.Verbose then
         names[current] = names[current] or 'Unnamed ' .. tostring(current):sub(8)
 
+        output_text_table = {}
+        table.insert(output_text_table, names[current])
+        if settings.VerboseKeys then
+            for key, val in pairs(current) do
+                if type(val) ~= 'string' then
+                    val = names[val] or 'Unnamed'
+                end
+                table.insert(output_text_table, key_combos[key] .. ': ' .. val)
```

Comment:
If you made the variable list, as described above, you can do `output_text_names:add(key_combos[key] .. ': ' .. val)` here.

Reply:
When I try to do this, I get a runtime error:

```
yush: Lua runtime error: yush/yush.lua:87: attempt to call method 'add' (a nil value)
```

This goes the same for when I tried it on the first insert of `names[current]`. I'm not very familiar with Lua so this could easily be a lack of understanding. I've renamed the variable and made it local, as mentioned above.

Reply:
Did you assign it via `= L{}` (instead of `= {}`)? That's the key part.

Reply:
Yeah. here's what I currently have:

```
local output_text_names = L{}
        output_text_names:add(names[current])
        if settings.VerboseKeys then
            for key, val in pairs(current) do
                if type(val) ~= 'string' then
                    val = names[val] or 'Unnamed'
                end
                output_text_names:add(key_combos[key] .. ': ' .. val)
            end
        end
        local output_text = output_text_names:concat('\n')
```

Also, what is the L{} thing actually called? I've been trying to do some searching for it and the S{} assignment, and the closest I can tell is S{} is a set and L{} is a list, but I haven't had any luck confirming.

Reply:
Found it. The method is `append`.

Reply:
Correct, this was my mistake (both now, and when I "designed" that API...). And yes, `L{}` is a list, i.e. for a contiguous, ordered collection of elements, and `S{}` for a set, i.e. unordered, unique collection of elements. The respective files are `addons/libs/lists.lua` and `addons/libs/sets.lua`.